### PR TITLE
Re-organize identifier selectors

### PR DESCRIPTION
**Pull request by gnidan**
Ref: #66 

- Add selector `data.views.decoder`, which returns a function `(definition, ref) => value`, to allow lookup of arbitrary data references with arbitrary AST nodes.

- Divide `data.current.identifiers` into a bunch of separately useful selectors:
  - `data.current.identifiers.definitions` which gives the AST definitions for current identifiers
  - `data.current.identifiers.refs` which gives the raw reference values (e.g. `{stack: 2}`) for current identifiers
  - `data.current.identifiers.decoded` renamed from `data.current.identifiers._`

- Rewrite reading from storage and parts of storage dereferencing to fix array child indexing and hopefully make things a bit clearer.

- Add integration tests for data decoding

### CODE REVIEW

File: lib/data/decode/utils.js [hunk 1]
```diff
@@ -7,6 +7,27 @@ import Web3 from "web3";
...
+    return Object.assign(
+      {}, ...Object.entries(value)
+        .map( ([key, inner]) => ({ [key]: cleanBigNumbers(inner) }) )
+    );
+
+  } else {
+    return value;
+  }
+}
+
```
Comment (cgewecke) [inline lib/data/decode/utils.js]: Nice.

Comment (cgewecke) [reply lib/data/decode/utils.js]: Oh you just moved it . .

File: test/data/decode/decoding.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+  string shortString;
+  string longString;
+
+  uint[] multipleFullWordArray;  // length 3, will take up 3 words
+  uint16[] withinWordArray;  // length 10, will take up >1/2 word
+  uint64[] multiplePartWordArray; // length 9, will take up 2.25 words
+
+  uint240[] inconvenientlyWordOffsetArray; // length 3, takes ~2.8 words
+
```
Comment (cgewecke) [inline test/data/decode/decoding.js]: Wow love this explanation.

File: test/data/decode/decoding.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+`;
+
+function imitateRun() {
+  let multipleFullWordArray = [];
+  let withinWordArray = [];
+  let multiplePartWordArray = [];
+  let inconvenientlyWordOffsetArray = [];
+  let shortString = "hello world";
+  let longString = "solidity storage is a fun lesson in endianness";
+
```
Comment (cgewecke) [inline test/data/decode/decoding.js]: Suspect `longString === false` here, not sure.

Comment (gnidan) [reply test/data/decode/decoding.js]: You know, `fun` in the dwarf-fortress sense of the word http://dwarffortresswiki.org/index.php/Fun

**Comment by gnidan** [comment]
Thank you sir!

File: test/data/decode/decoding.js (hunk 3 above)
Comment (cgewecke) [reply test/data/decode/decoding.js]: LOL!

**Pull request merged**